test/internal/helpers/teardown.go

```diff
@@ -12,27 +13,31 @@ import (
 )
 
 // TeardownCluster dumps the diagnostics from the test cluster if the test failed
-// and performs a cluster teardown.
+// and performs a cluster removal.
 func TeardownCluster(ctx context.Context, t *testing.T, cluster clusters.Cluster) {
 	t.Helper()
 
-	const (
-		environmentCleanupTimeout = 10 * time.Minute
-	)
+	dumpDiagnosticsIfFailed(ctx, t, cluster)
+	const environmentCleanupTimeout = 10 * time.Minute
+	ctx, cancel := context.WithTimeout(ctx, environmentCleanupTimeout)
+	defer cancel()
 
-	DumpDiagnosticsIfFailed(ctx, t, cluster)
+	t.Logf("INFO: cluster %s is being deleted\n", cluster.Name())
```

Comment:
I believe we can get rid of this one as well since we're logging the same message in `RemoveCluster`